### PR TITLE
SISRP-13892 Make RelatedCacheKeyTracker aware of JSON keys

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.berkeley.edu/browse/SISRP-13892

Refactor to make Cache::RelatedCacheKeyTracker work with Cache::JsonAddedCacher. Also changes RelatedCacheKeyTracker to group related keys by uid+class instead of just uid, so that expiring one set of related keys doesn't expire all other related keys for the user.

### CODE REVIEW

**Comment by paulkerschen** [comment]
Half of Travis is still upset about textbooks.

File: lib/cache/related_cache_key_tracker.rb [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+    end
+
     def self.included base
       base.extend ClassMethods
     end
 
     module ClassMethods
       def user_key(uid)
-        "related-cache-keys-#{uid}"
+        cache_key "related-cache-keys-#{uid}"
```
Comment (paulkerschen) [inline lib/cache/related_cache_key_tracker.rb]: This is the bit that adds the class name to the related-key collection.

File: lib/cache/json_added_cacher.rb [hunk 2]
```diff
@@ -16,6 +16,10 @@ def get_feed_as_json(force_cache_write=false)
       end
     end
 
+    def extended_instance_keys
+      super << self.class.json_key(instance_key)
+    end
```
Comment (paulkerschen) [inline lib/cache/json_added_cacher.rb]: Written this way to make it easier in the future if we have more work with multiply cached models.

**Comment by raydavis** [comment]
Nice labyrinth threading.

**Pull request merged**